### ISSUE TITLE
ISSUE: Cannot publicly access created gif

### ISSUE THREAD

**Issue by ab3lson** [moralmunky/Home-Assistant-Mail-And-Packages#411]
**Describe the bug**
I just updated from an older version of the custom component where I was previously getting the gif of my mail from a "camera" feed in home assistant. It looks like that is not possible in the newest revision and you have to use a randomized file path if you want external access for notifications. The issue I am having is that the URL given by the sensor.mail_image_url entity does not allow for external access. As a result, I receive a 404 error on my notifications. 

In previous versions, the user could provide a file path for the created gif. This was nice, because I could make sure that the directory was publicly accessible. Currently, it seems that the the random file name stored in sensor.mail_image_system_path does not get placed where sensor.mail_image_url points if the "allow external access" box is checked. Can this be fixed?

**Environment (please complete the following information):**
 - OS: HassOS
 - Type of device: RPi 4 2GB
 - Home Assistant version: Latest
 - Component version: Latest

**Logs**
2021-03-20 20:01:48 DEBUG (MainThread) [custom_components.mail_and_packages.sensor] Updating system image path to: images/mail_and_packages/

**Comment by firstof9** [comment]
Please [enable debugging](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/wiki/Troubleshooting#enable-debugging) and post the Home Assistant log that gets generated.

**Comment by moralmunky** [comment]
> Component version: Latest

You only indicate using the 'latest' version and you mention the setting is named "allow external access" so I assume you are on the 0.3.1 release and not the 0.3.2-b3 pre-release.

> Currently, it seems that the the random file name stored in sensor.mail_image_system_path does not get placed where sensor.mail_image_url points if the "allow external access" box is checked.

In 0.3.0 - 0.3.1 release  the `allow external access` changes the place the GIF is saved and changes the paths for the two sensors that provide a pathway to the image.

With `allow external access` turned on you should see the image in the config_path/www/mail_and_packages/random_name.gif

What path is the sensor.mail_image_system_path and what path do you find the image being stored in?

And yes, please provide the logs after turning on debugging.

**Comment by ab3lson** [comment]
Sorry I wasn't more specific with the version, I didn't know that there were beta versions. I am on release 0.3.1.
sensor.mail_image_system_path contains /config/images/mail_and_packages/67c04538-0356-45b4-836f-23ffa2d144ea.gif and sensor.mail_image_url contains https://DOMAIN/local/mail_and_packages/67c04538-0356-45b4-836f-23ffa2d144ea.gif. I am assuming that the gif at the image_system_path is supposed to be copied into the image_url path, but it does not appear there. When I enable debugging, that log that I posted in my original comment is the only log that I can see. Unless there is a way to filter out the rest of the Hass logs?

**Comment by firstof9** [comment]
`2021-03-20 20:01:48 DEBUG (MainThread) [custom_components.mail_and_packages.sensor] Updating system image path to: images/mail_and_packages/` would indicate that `Allow external access` has not been enabled in the options as it should report `www/mail_and_packages/`. 

Please double check the settings by clicking `Options` on Mail and Packages from the `Integrations` menu in Home Assistant.

**Comment by ab3lson** [comment]
I decided to delete the mail and packages folder in custom components and reinstall on more time and it appears to be working as expected now. One thing I did differently before the reinstall was make sure that the directory /config/www/mail_and_packages existed. It is possible that the directory was not created the first time around and was causing issues. I'll watch it over the next couple of days and make sure that it continues to behave as expected. Thanks for the quick responses!
